lib/chef/provider.rb

```diff
@@ -376,7 +376,18 @@ module ClassMethods
         # compile the resources, converging them, and then checking if any
         # were updated (and updating new-resource if so)
         def action(name, &block)
-          define_method("action_#{name}") { compile_and_converge_action(&block) }
```

Comment:
Yeah I was trying to understand what the magic was here.  So you simply have to wrap the block with a define_method and nothing else...